### PR TITLE
Replace usages of `Geometry.vertices` with appropriate fields on each subclass

### LINKED ISSUE #1736: Replace uses of `Geometry.vertices` with appropriate fields on each subclass
- [ ] Ensure polygons, multipolygons, and points are rendering

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1736

Verified the changes locally that bounds are getting computed correctly and polyline/polygons are being rendered as well.

@gino-m @JSunde   PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/map/gms/renderer/PolylineRenderer.kt [hunk 1]
```diff
@@ -33,16 +38,26 @@ class PolylineRenderer(
...
   private val polylines: MutableMap<Feature, MutableList<Polyline>> = HashMap()
 
-  override fun addFeature(feature: Feature, geometry: Geometry) {
-    val points = geometry.vertices
+  override fun addFeature(feature: Feature) {
+    when (feature.geometry) {
+      is Point,
+      is MultiPolygon,
+      is Polygon ->
+        throw IllegalArgumentException("Can't render polyline for geometry: ${feature.geometry}")
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/map/gms/renderer/PolylineRenderer.kt:46]: Perhaps these should be covered by `else`? I'd also update the error to something more descriptive, e.g.

`error("PolylineRendered expected LineString or LinearRing, but got ${feature.geometry::class.simpleName}")`

**Review by gino-m — CHANGES_REQUESTED** [review]
Thanks for this! A few suggestions

File: ground/src/main/java/com/google/android/ground/ui/map/gms/renderer/PolylineRenderer.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/renderer/PolylineRenderer.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/map/gms/GmsExt.kt [hunk 2]
```diff
@@ -23,18 +23,15 @@ import com.google.android.ground.ui.map.Bounds
 /** Extensions for indirectly using GMS functions in map-provider agnostic codebase. */
 object GmsExt {
 
-  fun Bounds.contains(coordinates: Coordinates): Boolean =
-    toGoogleMapsObject().contains(coordinates.toGoogleMapsObject())
-
   fun Bounds.contains(geometry: Geometry): Boolean {
     val latLngBounds = toGoogleMapsObject()
-    return geometry.vertices.any { latLngBounds.contains(it.toLatLng()) }
+    return geometry.toLatLngList().any { latLngBounds.contains(it) }
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/GmsExt.kt]: Done

File: ground/src/main/java/com/google/android/ground/persistence/remote/firebase/schema/LoiMutationConverter.kt [hunk 3]
```diff
@@ -41,10 +42,11 @@ internal object LoiMutationConverter {
     map[LoiConverter.SUBMISSION_COUNT] = mutation.submissionCount
 
     when (val geometry = mutation.geometry) {
-      is Point -> map.addGeometryCoordinates(toGeoPoint(geometry), LoiConverter.POINT_TYPE)
+      is Point ->
+        map.addGeometryCoordinates(geometry.coordinates.toGeoPoint(), LoiConverter.POINT_TYPE)
       is Polygon ->
         map.addGeometryCoordinates(
-          toGeoPointList(geometry.shell.vertices),
+          geometry.shell.coordinates.toGeoPointList(),
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/persistence/remote/firebase/schema/LoiMutationConverter.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/polygon/PolygonDrawingTaskFragment.kt [hunk 4]
```diff
@@ -74,7 +74,7 @@ class PolygonDrawingTaskFragment : Hilt_PolygonDrawingTaskFragment<PolygonDrawin
   }
 
   private fun onFeatureUpdated(feature: Feature?) {
-    val vertexCount = feature?.geometry?.size ?: 0
+    val vertexCount = feature?.geometry?.size() ?: 0
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/polygon/PolygonDrawingTaskFragment.kt]: Thanks for the suggestion! I didn't notice that the size is only being used for determining non-zero vertex count. Done

**Comment by shobhitagarwal1612** [comment]
@gino-m PTAL

**Review by gino-m — APPROVED** [review]
LGTM, thanks again!

**Pull request merged**